src/Microsoft.Sbom.Api/Workflows/Helpers/IJsonArrayGenerator.cs

```diff
@@ -1,25 +1,21 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Collections.Generic;
 using System.Threading.Tasks;
-using Microsoft.Sbom.Extensions;
+using Microsoft.Sbom.Extensions.Entities;
 
 namespace Microsoft.Sbom.Api.Workflows.Helpers;
 
 /// <summary>
-/// Used to generate array objects in the JSON serializer.
+/// Used to generate array objects to be written to the JSON serializer.
 /// </summary>
 public interface IJsonArrayGenerator<T>
     where T : IJsonArrayGenerator<T>
 {
-    public ISbomConfig SbomConfig { get; set; }
-
-    public string SpdxManifestVersion { get; set; }
-
     /// <summary>
-    /// Generates an array in the json serializer with the headerName and writes all elements of the
-    /// specific type into the array.
+    /// Generates all the JSON objects that need to be written to the SBOM.
     /// </summary>
-    /// <returns>The list of failures.</returns>
-    public Task<GenerationResult> GenerateAsync();
+    /// <returns>GenerationResult with objects to write to the SBOM and failures.</returns>
+    public Task<GenerationResult> GenerateAsync(IList<ManifestInfo> manifestInfosFromConfig, HashSet<string> elementsSpdxIdList);
```

Comment:
Generics (and propagate)
```suggestion
    public Task<GenerationResult> GenerateAsync(IEnumerable<ManifestInfo> manifestInfosFromConfig, ISet<string> elementsSpdxIdList);
```